app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PostsController < ApplicationController
+  before_action :get_post, only: [:show, :destroy, :edit, :update]
+  before_action :get_comments, only: [:show]
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def show
+  end
```

Comment:
Just fyi, you can omit empty actions and they'll still execute if there's a view present. But leaving them in can be a nice visual cue to the reader that they exist - without having to go checking for views.